test/dotty/partest/DPCompilerTest.scala

```diff
@@ -0,0 +1,266 @@
+package dotty.partest
+
+import org.junit.runner.RunWith
+import scala.reflect.io.{ Path, Directory, File => SFile }
+import java.io.{ File => JFile }
+import scala.tools.partest.nest.FileManager
+
+
+/** Test classes annotated with this category won't be run by sbt JUnit test,
+ * because JUnit has a testOption in the Build.scala file which excludes this
+ * category. Always together with DPCompilerTest! */
+trait NonPartestTarget
+trait PartestTarget
+
+/** Trait to be mixed into a subclass of a test to create a version that is run
+  * by partest. Always together with NonPartestTarget!
+  *
+  * This trait is a bridge between the JUnit CompilerTest and partest. It
+  * generates the necessary structure for partest to run on directory
+  * DPConfig.testRoot. It does not run partest, sbt does.
+  *
+  * A test can either be a file or a directory. The test is in a parent
+  * directory that determines the kind of test:
+  * - pos: checks that compilation succeeds
+  * - neg: checks that compilation fails with the given number of errors
+  * (- run: compilation succeeds and running generates the given output.)
+  * Compiler flags and the number of errors expected from a neg test are read
+  * from test.flags and test.nerr files.
+  */
+@RunWith(value = classOf[DPPrepJUnitRunner])
+trait DPCompilerTest extends test.CompilerTest {
```

Comment:
DottyPartest... Should I spell it out for all of these classes?

On Thu, Apr 16, 2015 at 3:26 PM, Dmitry Petrashko <notifications@github.com>
wrote:

> In test/dotty/partest/DPCompilerTest.scala
> <https://github.com/lampepfl/dotty/pull/482#discussion_r28508516>:
>
> > +  * by partest. Always together with NonPartestTarget!
> > +  *
> > +  * This trait is a bridge between the JUnit CompilerTest and partest. It
> > +  * generates the necessary structure for partest to run on directory
> > +  * DPConfig.testRoot. It does not run partest, sbt does.
> > +  *
> > +  * A test can either be a file or a directory. The test is in a parent
> > +  * directory that determines the kind of test:
> > +  * - pos: checks that compilation succeeds
> > +  * - neg: checks that compilation fails with the given number of errors
> > +  * (- run: compilation succeeds and running generates the given output.)
> > +  * Compiler flags and the number of errors expected from a neg test are read
> > +  * from test.flags and test.nerr files.
> > +  */
> > +@RunWith(value = classOf[DPPrepJUnitRunner])
> > +trait DPCompilerTest extends test.CompilerTest {
>
> What does DP stand for?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lampepfl/dotty/pull/482/files#r28508516>.
>